lib/passport-configurator.js

```diff
@@ -300,15 +300,21 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
                 }
               } else {
                 user.hasPassword(password, function(err, ok) {
-                  if (ok) {
-                    done(null, userProfile);
+                  if (ok) {          
```

Comment:
I will look into that. I don't think it is possible in the current case since email not valid is basically within the OK success. It means user/pass are fine but email is still NOT Valid